### PR TITLE
Switch to YAML-based configuration file

### LINKED ISSUE #70: Switch to YAML configuration
Moved from https://github.com/osiam/resource-server/issues/44 & https://github.com/osiam/auth-server/issues/40:

Using properties files is not really an optimal way to configure this application. As discussed [here](https://github.com/osiam/addon-self-administration/issues/123), we decided to switch to YAML.

### PR DESCRIPTION
**Pull request by tkrille**
Load configuration file via `spring.config.location` property to replace
usage of `@PropertySource`.
The only way to always define it on startup is using default properties.

Remove configuration property `org.osiam.home`, because it's not used
anymore.

Refactor LDAP configuration to support lists and hashes for DN
patterns and SCIM-to-LDAP mapping.
Use `@ConfigurationProperties` to bind to these complex attributes.

Remove configuration property `org.osiam.ldap.server.groupsearchbase`,
because it's not documented and apparently has no effect.
Replace its usage with its default value: the empty string.

Resolves #70
Related to #35

### CODE REVIEW

**Comment by tkrille** [comment]
Related changes:
- ITs: https://github.com/osiam/connector4java-integration-tests/pull/317
- Docker image: https://github.com/osiam/docker-image/pull/32

File: src/main/java/org/osiam/configuration/LdapAuthentication.java [hunk 1]
```diff
@@ -37,45 +37,43 @@
 import org.springframework.security.ldap.authentication.LdapAuthenticator;
 import org.springframework.security.ldap.userdetails.DefaultLdapAuthoritiesPopulator;
 
+import javax.annotation.PostConstruct;
```
Comment (wallner) [inline src/main/java/org/osiam/configuration/LdapAuthentication.java:40]: Codacy Issue found: [Avoid unused imports such as 'javax.annotation.PostConstruct'](https://www.codacy.com/app/OSIAM/osiam/file/1888507949/issues/source?bid=3080997&fileBranchId=3092593#l40)

File: src/main/java/org/osiam/configuration/LdapAuthentication.java [hunk 2]
```diff
@@ -37,45 +37,43 @@
...
+import java.util.Arrays;
```
Comment (wallner) [inline src/main/java/org/osiam/configuration/LdapAuthentication.java:41]: Codacy Issue found: [Avoid unused imports such as 'java.util.Arrays'](https://www.codacy.com/app/OSIAM/osiam/file/1888507949/issues/source?bid=3080997&fileBranchId=3092593#l41)

**Comment by wallner** [comment]
Thank you.

**Pull request merged**